src/shogun/machine/Machine.cpp

```diff
@@ -176,3 +176,43 @@ std::shared_ptr<LatentLabels> Machine::apply_latent(std::shared_ptr<Features> da
 	error("This machine does not support apply_latent()");
 	return NULL;
 }
+
+std::shared_ptr<Machine> Machine::fit(
+    const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& lab)
+{
+	require(data, "No features provided");
+	require(lab, "No labels provided");
+	require(
+	    lab->get_num_labels() == data->get_num_vectors(),
+	    "Number of training vectors does not match number of labels");
+	lab->ensure_valid(get_name());
+	auto sub = connect_to_signal_handler();
+	if (support_feature_dispatching())
+	{
+		if (support_dense_dispatching() && data->get_feature_class() == C_DENSE){
+			train_dense(data);
+		}
+		else if (
+		    support_string_dispatching() &&
+		    data->get_feature_class() == C_STRING){
+				train_string(data);
+			}	
+		else{
+			error("Training with {} is not implemented!", data->get_name());
+			return nullptr;
+		}	
+	}
+	else{
+		fit_machine(data, lab);
+	}
+	sub.unsubscribe();
+	reset_computation_variables();
+	return shared_from_this()->as<Machine>();
+}
+
+std::shared_ptr<Labels>
+Machine::predict(const std::shared_ptr<Features>& data)
+{
+	require(data, "No labels provided");
```

Comment:
as above, wouldnt add this method, it just adds noise ... no need. We can rename the methods later if we want